WalletWasabi/WabiSabi/Client/WabiSabiHttpApiClient.cs

```diff
@@ -67,15 +71,52 @@ public async Task<RoundState[]> GetStatusAsync(CancellationToken cancellationTok
 		public Task ReadyToSignAsync(ReadyToSignRequestRequest request, CancellationToken cancellationToken) =>
 			SendAndReceiveAsync<ReadyToSignRequestRequest>(RemoteAction.ReadyToSign, request, cancellationToken);
 
+		private async Task<HttpResponseMessage> SendWithRetriesAsync(RemoteAction action, string jsonString, CancellationToken cancellationToken)
+		{
+			var errors = new List<Exception>();
+
+			var start = DateTime.Now;
+
+			for (var attempt = 0; attempt < MaxRetries; attempt++)
+			{
+				try
+				{
+					using StringContent content = new(jsonString, Encoding.UTF8, "application/json");
+
+					// Any transport layer errors will throw an exception here.
+					var response = await _client.SendAsync(HttpMethod.Post, GetUriEndPoint(action), content, cancellationToken).ConfigureAwait(false);
+
+					var totalTime = DateTime.Now - start;
+
+					if (errors.Any())
+					{
+						Logger.LogDebug($"Received a response for {action} in {totalTime.TotalSeconds:0.##s} after {attempt} failed attempts: {new AggregateException(errors)}.");
+					}
+					else
+					{
+						Logger.LogDebug($"Received a response for {action} in {totalTime.TotalSeconds:0.##s}.");
+					}
+
+					return response;
+				}
+				catch (Exception e)
+				{
+					errors.Add(e);
+				}
+			}
+
+			throw new AggregateException(errors);
+		}
+
 		private async Task<string> SendAsync<TRequest>(RemoteAction action, TRequest request, CancellationToken cancellationToken) where TRequest : class
```

Comment:
Personally, I would rename this method as well because `SendAsync` typically means "just once".

Reply:
name both `SendWithRetriesAsync`, so that the `string` variant is an overload, or did you have something else in mind?

Reply:
That's what I had in mind 👍